From 71439b87f51b48d85ff83312c2dd0ea5179b799f Mon Sep 17 00:00:00 2001
From: Gilles DOFFE <g.doffe@gmail.com>
Date: Mon, 14 Aug 2023 03:41:03 +0200
Subject: [PATCH 3/7] boards/nucleo64: clean motor_driver old configuration

The motor_driver module has been reworked in a previous commit to be
compliant with RIOT device driver guide.
Thus declaration in board.h is no  more needed and should not work
anymore.

Signed-off-by: Gilles DOFFE <g.doffe@gmail.com>
---
 boards/common/nucleo64/include/board.h | 32 --------------------------
 boards/nucleo-f446re/Makefile.features |  1 -
 2 files changed, 33 deletions(-)

diff --git a/boards/common/nucleo64/include/board.h b/boards/common/nucleo64/include/board.h
index 7aedd553e6..c917f3cfd3 100644
--- a/boards/common/nucleo64/include/board.h
+++ b/boards/common/nucleo64/include/board.h
@@ -24,7 +24,6 @@
 
 #include "board_nucleo.h"
 #include "arduino_pinmap.h"
-#include "motor_driver.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -56,37 +55,6 @@ extern "C" {
 #endif
 /** @} */
 
-/**
- * @name Describe DC motors with PWM channel and GPIOs
- * @{
- */
-static const motor_driver_config_t motor_driver_config[] = {
-    {
-        .pwm_dev         = 1,
-        .mode            = MOTOR_DRIVER_1_DIR,
-        .mode_brake      = MOTOR_BRAKE_HIGH,
-        .pwm_mode        = PWM_LEFT,
-        .pwm_frequency   = 20000U,
-        .pwm_resolution  = 2250U,
-        .nb_motors       = 1,
-        .motors          = {
-            {
-                .pwm_channel            = 0,
-                .gpio_enable            = 0,
-                .gpio_dir0              = ARDUINO_PIN_15,
-                .gpio_dir1_or_brake     = 0,
-                .gpio_dir_reverse       = 0,
-                .gpio_enable_invert     = 0,
-                .gpio_brake_invert      = 0,
-            },
-        },
-        .cb = NULL,
-    },
-};
-
-#define MOTOR_DRIVER_NUMOF           ARRAY_SIZE(motor_driver_config)
-/** @} */
-
 /**
  * @name Describe MRF24J40 radio
  * @{
diff --git a/boards/nucleo-f446re/Makefile.features b/boards/nucleo-f446re/Makefile.features
index da373b85b3..c7599b2f3f 100644
--- a/boards/nucleo-f446re/Makefile.features
+++ b/boards/nucleo-f446re/Makefile.features
@@ -14,7 +14,6 @@ FEATURES_PROVIDED += periph_uart
 FEATURES_PROVIDED += periph_qdec
 
 # Various other features (if any)
-FEATURES_PROVIDED += motor_driver
 FEATURES_PROVIDED += riotboot
 
 # load the common Makefile.features for Nucleo boards
-- 
2.37.0

